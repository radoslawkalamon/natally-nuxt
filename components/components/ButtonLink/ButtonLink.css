.button-link {
  background-color: var(--color-accent);
  border-radius: var(--border-radius);
  box-shadow: inset 0 0 0 var(--border-size) var(--color-shadow), 0 var(--border-size) 0 var(--color-shadow);
  color: var(--color-off-white-1);
  display: inline-block;
  font-family: var(--font-family-sans);
  font-size: var(--font-size-base);
  font-weight: bold;
  margin: 0;
  padding: 0.5em 1em;
  text-decoration: none;
  transition: box-shadow var(--time-animation) ease, opacity var(--time-animation) ease;
}

.button-link:hover {
  opacity: 0.9;
  text-decoration: underline;
}

.button-link:active {
  box-shadow: inset 0 0 0 var(--border-size) var(--color-shadow), 0 0 0 var(--color-shadow);
  opacity: 0.75;
}
