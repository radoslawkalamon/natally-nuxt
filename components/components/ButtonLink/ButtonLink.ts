import Vue from 'vue'

export default Vue.extend({
  name: 'ComponentsButtonLink',
  inheritAttrs: false,
  props: {
    label: {
      type: String,
      required: true
    }
  }
})
