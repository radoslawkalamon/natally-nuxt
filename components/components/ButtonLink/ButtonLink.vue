<template>
  <NuxtLink
    class="button-link"
    data-test="components-button-link"
    v-text="label"
  />
</template>

<script src="./ButtonLink.ts" lang="ts"></script>
<style src="./ButtonLink.css" scoped></style>
