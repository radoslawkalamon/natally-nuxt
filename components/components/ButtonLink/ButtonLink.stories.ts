import { Meta, Story } from '@storybook/vue'
import ButtonLink from './ButtonLink.vue'

const meta: Meta = {
  title: 'Components / Button Link',
  component: ButtonLink
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: { ButtonLink },
  props: Object.keys(argTypes),
  template: '<ButtonLink v-bind="$props" />'
})

export const Default: Story = Template.bind({})
Default.args = {
  label: 'Click here'
}
