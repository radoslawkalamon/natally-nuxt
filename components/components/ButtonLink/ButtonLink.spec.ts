import ButtonLink from '@/components/components/ButtonLink/ButtonLink.vue'
import { shallPassUnitSanityTest } from '@/devtools/jest.common.spec'

describe('Components / Button Link', () => {
  describe('Unit', () => {
    shallPassUnitSanityTest({
      component: ButtonLink,
      options: {
        propsData: {
          label: 'Item label'
        }
      }
    })
  })
})
