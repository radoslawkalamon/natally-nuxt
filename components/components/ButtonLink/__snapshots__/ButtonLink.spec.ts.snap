// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Button Link Unit shall pass unit sanity test 1`] = `"<nuxtlink-stub class=\\"button-link\\">Item label</nuxtlink-stub>"`;
